### PR TITLE
NETBEANS-73: Autodetect id_rsa and id_dsa keys for Git private key.

### PR DESCRIPTION
**Pull request by lkishalmi**
Simple patch for https://issues.apache.org/jira/browse/NETBEANS-73

### CODE REVIEW

File: git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java [hunk 1]
```diff
@@ -686,6 +686,7 @@ public SSHConnectionSettingsType () {
                 settingsPanel.savePasswordCheckBox
             };
             acceptableSchemes = EnumSet.of(Scheme.SSH, Scheme.SFTP);
+            settingsPanel.txtIdentityFile.setText(getDefaultIdentityFilePath());
```
Comment (emilianbold) [inline git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java]: This line does not seem necessary. `SSHConnectionSettingsType.populateFields` does call

```java
                String identityFile = getDefaultIdentityFilePath();
                settingsPanel.txtIdentityFile.setText(identityFile);
```

File: git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java [hunk 2]
```diff
@@ -853,9 +854,13 @@ protected int getPreferedPanelHeight () {
...
-            if (!Utilities.isWindows()) {
-                identityFile = System.getProperty("user.home") + File.separator //NOI18N
-                        + ".ssh" + File.separator + "id_dsa"; //NOI18N
+            File sshDir = new File(System.getProperty("user.home"), ".ssh"); //NOI18N
+            File rsaKey = new File(sshDir, "id_rsa"); //NOI18N
+            File dsaKey = new File(sshDir, "id_dsa"); //NOI18N
+            if (rsaKey.canRead()) {
+                identityFile = rsaKey.getAbsolutePath();
+            } else if (dsaKey.canRead()) {
+                identityFile = dsaKey.getAbsolutePath();
```
Comment (emilianbold) [inline git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java]: This looks OK.

File: git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java (hunk 1 above)
Comment (lkishalmi) [reply git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java]: Unfortunately populateFields is not called in every case. Most importantly when the repository URL is being copied from github/gitlab where the ssh:// prefix is just missing from the URL. Probably I could raise a bug on that. And mark this line as a workaround in the code, till it gets fixed. I feel the fix would be a bit out of the trivial category.

Comment (emilianbold) [reply git/src/org/netbeans/modules/git/ui/repository/remote/RemoteRepository.java]: But what if I'm not using an identity file?

**Comment by lkishalmi** [comment]
I've made a small change to do the detection only when the key auth is selected.

**Comment by emilianbold** [comment]
Committed: https://github.com/apache/incubator-netbeans/commit/ff5f3c4e175cf15c1b4b0caebfb5047be4196a44

**Comment by emilianbold** [comment]
Please close this PR.

**Comment by lkishalmi** [comment]
Thank you!

**Pull request closed without merging**